packages/opentelemetry-plugin-grpc/src/grpc.ts

```diff
@@ -122,24 +127,28 @@ export class GrpcPlugin extends BasePlugin<grpc> {
     }
   }
 
-  private _getSpanContext(metadata: grpcTypes.Metadata): SpanContext | null {
-    const metadataValue = metadata.getMap()[GRPC_TRACE_KEY] as Buffer;
-    // Entry doesn't exist
-    if (!metadataValue) {
-      return null;
-    }
-    return this._tracer.getBinaryFormat().fromBytes(metadataValue);
+  private _getSpanContext(
+    metadata: grpcTypes.Metadata
+  ): SpanContext | undefined {
+    return getExtractedSpanContext(
+      this._tracer.getHttpTextFormat().extract(Context.TODO, metadata.getMap())
+    );
   }
 
   private _setSpanContext(
     metadata: grpcTypes.Metadata,
     spanContext: SpanContext
   ): void {
-    const serializedSpanContext = this._tracer
-      .getBinaryFormat()
-      .toBytes(spanContext);
-    const buffer = Buffer.from(serializedSpanContext);
-    metadata.set(GRPC_TRACE_KEY, buffer);
+    const carrier = {};
+    this._tracer
+      .getHttpTextFormat()
+      .inject(
+        setExtractedSpanContext(Context.ROOT_CONTEXT, spanContext),
+        carrier
+      );
+    for (const [k, v] of Object.entries(carrier)) {
```

Comment:
I'm going to approve this for now, but we should be using the new `setter` interface for propagators to handle this. https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-propagators.md#setter-argument